Assets/Mirror/Runtime/NetworkBehaviour.cs

```diff
@@ -418,6 +418,79 @@ protected NetworkIdentity GetSyncVarNetworkIdentity(uint netId, ref NetworkIdent
             return identityField;
         }
 
+        // helper function for [SyncVar] NetworkIdentities.
+        protected void SetSyncVarNetworkBehaviour<T>(T newBehaviour, ref T behaviourField, ulong dirtyBit, ref SyncNetBehaviour syncField) where T : NetworkBehaviour
+        {
+            if (getSyncVarHookGuard(dirtyBit))
+                return;
+
+            uint newNetId = 0;
+            if (newBehaviour != null)
+            {
+                newNetId = newBehaviour.netId;
+                if (newNetId == 0)
+                {
+                    logger.LogWarning($"{nameof(SetSyncVarNetworkBehaviour)} NetworkIdentity " + newBehaviour + " has a zero netId. Maybe it is not spawned yet?");
+                }
+            }
+
+            // todo update this log for NetworkBehaviours
+            if (logger.LogEnabled()) logger.Log("SetSyncVarNetworkIdentity NetworkIdentity " + GetType().Name + " bit [" + dirtyBit + "] netIdField:" + syncField + "->" + newNetId);
+            SetDirtyBit(dirtyBit);
+
+            int compIndex = newBehaviour.ComponentIndex;
+            syncField = new SyncNetBehaviour(newNetId, compIndex);
+
+            // assign new one on the server, and in case we ever need it on client too
+            behaviourField = newBehaviour;
+        }
+
+        // helper function for [SyncVar] NetworkIdentities.
+        // -> ref GameObject as second argument makes OnDeserialize processing easier
+        protected T GetSyncVarNetworkBehaviour<T>(SyncNetBehaviour syncNetBehaviour, ref T behaviourField) where T : NetworkBehaviour
+        {
+            // server always uses the field
+            if (isServer)
+            {
+                return behaviourField;
+            }
+
+            // client always looks up based on netId because objects might get in and out of range
+            // over and over again, which shouldn't null them forever
+
+            bool found = NetworkIdentity.spawned.TryGetValue(syncNetBehaviour.netId, out NetworkIdentity identity);
+            if (!found)
```

Comment:
'found' variable seems unnecessary. you could just check if !TryGetValue then return